nix/shell.sh

```diff
@@ -0,0 +1,105 @@
+echo nix/shell.sh: Entering a devShell
+export SRC_ROOT=$(pwd)
+export PATH=$SRC_ROOT/build/bin:$PATH
+
+banner()
+{
+    echo "+------------------------------------------+"
+    printf "| %-40s |\n" "$1"
+    echo "+------------------------------------------+"
+}
+
+
+function configure {
+    banner "Configuring with cmake"
+    cmake -S . -B./build \
+          -DBUILD_TESTING=ON \
+          -DS2N_INTEG_TESTS=ON \
+          -DS2N_INSTALL_S2NC_S2ND=ON \
+          -DS2N_FAST_INTEG_TESTS=ON \
+          -DBUILD_SHARED_LIBS=ON \
+          -DCMAKE_BUILD_TYPE=RelWithDebInfo
```

Comment:
This is fine for now but it's a bit limiting. I'm not sure if there's a better way to expose the matrix of options one might want to test, though.

Reply:
My interpretation here is that this is an opinionated default. If you want to build with your own options you can invoke CMake directly. Perhaps there is a limited set of default options that invoke CMake in a few different ways. But I don't think that exposing the whole matrix of options makes sense, CMake already does that and you can use if for that purpose.